### DISCUSSION TITLE
Proper way to handle errors from python restorecon binding?

### DISCUSSION THREAD

**Discussion by brianjmurrell** [SELinuxProject/selinux#434] [Q&A] [answered]
It's not clear to me in which ways the `restorecon()` python binding can return an error/failure?

Can it raise an exception or does it always simply report errors in it's return code?  I.e. I need to do:

```python

rc = selinux.restorecon(path)
if rc != 0:
    [error handling]
```

And/or do I need a `try`/`except` block around `restorecon()` as something below it could raise an exception?

I have seen reference to code that calls it as such:

```python

try:
  selinux.restorecon(path)
except selinux.SELinuxError:
  [error handling]
```

**Comment by bachradsusi** [comment, marked as answer]
https://lists.fedoraproject.org/archives/list/selinux@lists.fedoraproject.org/message/E2TGYSKXYATVRFR36PS6EWXGXQUHBAMW/
